File: backend/app/schemas/admin_dashboard.py
from __future__ import annotations

import uuid
from datetime import date, datetime

from pydantic import BaseModel


class DashboardActivityItem(BaseModel):
    id: uuid.UUID
    description: str
    source: str
    type: str
    amount: str
    created_at: datetime


class DashboardTeeOccupancy(BaseModel):
    booked_slots: int
    total_slots: int
    occupancy_pct: int | None


class DashboardNotice(BaseModel):
    code: str
    message: str


class DashboardTargetContext(BaseModel):
    """An active club target for the current period, surfaced alongside dashboard KPIs."""

    domain_key: str
    domain_label: str
    metric_key: str
    metric_label: str
    period_key: str
    period_start: date
    period_end: date
    target_value: float
    unit: str


class AdminDashboardSummaryResponse(BaseModel):
    member_count: int
    tee_occupancy: DashboardTeeOccupancy
    tee_warnings: list[DashboardNotice]
    recent_activity: list[DashboardActivityItem]
    active_targets: list[DashboardTargetContext] = []

File: frontend/src/types/admin-dashboard.ts
export interface DashboardActivityItem {
  id: string;
  description: string;
  source: string;
  type: string;
  amount: string;
  created_at: string;
}

export interface DashboardTeeOccupancy {
  booked_slots: number;
  total_slots: number;
  occupancy_pct: number | null;
}

export interface DashboardNotice {
  code: string;
  message: string;
}

export interface DashboardTargetContext {
  domain_key: string;
  domain_label: string;
  metric_key: string;
  metric_label: string;
  period_key: string;
  period_start: string;
  period_end: string;
  target_value: number;
  unit: string;
}

export interface AdminDashboardSummary {
  member_count: number;
  tee_occupancy: DashboardTeeOccupancy;
  tee_warnings: DashboardNotice[];
  recent_activity: DashboardActivityItem[];
  active_targets: DashboardTargetContext[];
  unpaid_bookings_today: number;
  no_show_risk_count: number;
  arrivals_due_count: number;
  close_day_ready: boolean;
}
